File: Data Structures & Algorithms/valid-palindrome-ii/submission-1.ts
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    validPalindrome(s: string): boolean {
        let chance = true;
        let chars = [...s.toLowerCase()].filter(isAlnum);
        let searches = [[0, chars.length - 1]];
        while (searches.length > 0) {
            let [i, j] = searches.pop();
            if (i >= j) {
                return true;
            }
            if (s[i] === s[j]) {
                searches.push([i + 1, j - 1]);
            } else {
                if (chance) {
                    searches.push([i + 1, j]);
                    searches.push([i, j - 1]);
                    chance = false;
                }
            }
        }
        return false;
    }
    
}
function isAlnum(ch: string): boolean {

  return /^[a-z0-9]$/i.test(ch);

}

File: Data Structures & Algorithms/word-break/submission-1.py
class Solution:
    def wordBreak(self, s: str, wordDict: List[str]) -> bool:
        n = len(s)
        d = [[(s[i: j] in wordDict) if j >= i else False for j in range(n+1)] for i in range(n)]
        searches = [0]
        searched = set()
        while searches:
            k = searches.pop()
            if k in searched:
                continue
            if k >= n:
                return True
            for j in range(n+1):
                if d[k][j]:
                    searches.append(j)
            searched.add(k)
        return False